# <center/> Visualizing Violence in Burkina Faso <center>

## Wrangling and Exploring

While we look for the data we really want we can make some observations and transform the data based on assumptions or theories.

In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

In [2]:
burkina_df = pd.read_csv('burkina_df.csv')
burkina_df.set_index('EVENT_DATE', inplace=True)
burkina_df.reset_index(inplace=True)
burkina_df.head()

EVENT_DATE       DISORDER_TYPE                  EVENT_TYPE  \
0  1997-02-01      Demonstrations                    Protests   
1  1997-02-01      Demonstrations                    Protests   
2  1997-02-04      Demonstrations                    Protests   
3  1997-02-04      Demonstrations                    Protests   
4  1998-12-13  Political violence  Violence against civilians   

              SUB_EVENT_TYPE                                   ACTOR1  \
0  Protest with intervention                Protesters (Burkina Faso)   
1  Protest with intervention                Protesters (Burkina Faso)   
2  Protest with intervention                Protesters (Burkina Faso)   
3  Protest with intervention                Protesters (Burkina Faso)   
4                     Attack  Unidentified Armed Group (Burkina Faso)   

             ASSOC_ACTOR_1  INTER1                                     ACTOR2  \
0  Students (Burkina Faso)       6  Police Forces of Burkina Faso (1987-2014)   
1  Students (Burkina Faso)       6  Police Forces of Burkina Faso (1987-2014)   
2  Students (Burkina Faso)       6  Police Forces of Burkina Faso (1987-2014)   
3  Students (Burkina Faso)       6  Police Forces of Burkina Faso (1987-2014)   
4                      NaN       3                   Civilians (Burkina Faso)   

                ASSOC_ACTOR_2  INTER2  INTERACTION  CIVILIAN_TARGETING  \
0                         NaN       1           16                 NaN   
1                         NaN       1           16                 NaN   
2                         NaN       1           16                 NaN   
3                         NaN       1           16                 NaN   
4  Journalists (Burkina Faso)       7           37  Civilian targeting   

      LOCATION  LATITUDE  LONGITUDE  \
0  Ouagadougou     12.37     -1.525   
1  Ouagadougou     12.37     -1.525   
2  Ouagadougou     12.37     -1.525   
3  Ouagadougou     12.37     -1.525   
4  Ouagadougou     12.37     -1.525   

                                               NOTES  FATALITIES  
0  Police fired teargas at protesters demonstrati...           0  
1  Students demontrated for the release of two st...           0  
2  Police fired teargas at student protesters dem...           0  
3  Police fired teargas at students protesting fo...           0  
4  Norbert Zongo, an outspoken journalist against...           4

In [3]:
burkina_df['EVENT_DATE'] = pd.to_datetime(burkina_df['EVENT_DATE'])

In [4]:
burkina_df.dtypes

EVENT_DATE            datetime64[ns]
DISORDER_TYPE                 object
EVENT_TYPE                    object
SUB_EVENT_TYPE                object
ACTOR1                        object
ASSOC_ACTOR_1                 object
INTER1                         int64
ACTOR2                        object
ASSOC_ACTOR_2                 object
INTER2                         int64
INTERACTION                    int64
CIVILIAN_TARGETING            object
LOCATION                      object
LATITUDE                     float64
LONGITUDE                    float64
NOTES                         object
FATALITIES                     int64
dtype: object

I want to really examine each column. I will be referncing the ACLED Codebook to make explanations or give descriptions

In [5]:
burkina_df['DISORDER_TYPE'].value_counts()

Political violence                    5341
Strategic developments                1841
Demonstrations                        1336
Political violence; Demonstrations       9
Name: DISORDER_TYPE, dtype: int64




If there's confusion between whether something is political violence or a

demonstration I'm going to assume there was political violence - especially 

when we only have 9 events. Let's lower our DISTORDER_TYPE to 3 unique values.

This will help us later when we prepare the data for predictive analysis. 

In [6]:
burkina_df['DISORDER_TYPE'] = burkina_df['DISORDER_TYPE'].str.replace('; Demonstrations', '')

In [7]:
burkina_df['DISORDER_TYPE'].value_counts()

Political violence        5350
Strategic developments    1841
Demonstrations            1336
Name: DISORDER_TYPE, dtype: int64

In [8]:
burkina_df['EVENT_TYPE'].value_counts()

Violence against civilians    2427
Strategic developments        1841
Battles                       1652
Explosions/Remote violence    1148
Protests                      1027
Riots                          432
Name: EVENT_TYPE, dtype: int64

In [9]:
burkina_df['SUB_EVENT_TYPE'].value_counts()

Attack                                 1709
Armed clash                            1633
Looting/property destruction           1261
Peaceful protest                        941
Abduction/forced disappearance          690
Air/drone strike                        684
Other                                   382
Remote explosive/landmine/IED           351
Violent demonstration                   318
Disrupted weapons use                   123
Mob violence                            114
Shelling/artillery/missile attack       107
Protest with intervention                77
Change to group/activity                 42
Sexual violence                          28
Arrests                                  15
Non-state actor overtakes territory      15
Non-violent transfer of territory        11
Excessive force against protesters        9
Agreement                                 7
Grenade                                   5
Government regains territory              4
Suicide bomb                    

In [10]:
burkina_df[burkina_df['DISORDER_TYPE'] == 'Demonstrations']

EVENT_DATE   DISORDER_TYPE EVENT_TYPE             SUB_EVENT_TYPE  \
0    1997-02-01  Demonstrations   Protests  Protest with intervention   
1    1997-02-01  Demonstrations   Protests  Protest with intervention   
2    1997-02-04  Demonstrations   Protests  Protest with intervention   
3    1997-02-04  Demonstrations   Protests  Protest with intervention   
5    1998-12-15  Demonstrations   Protests           Peaceful protest   
...         ...             ...        ...                        ...   
8431 2023-04-13  Demonstrations      Riots      Violent demonstration   
8440 2023-04-14  Demonstrations      Riots      Violent demonstration   
8453 2023-04-16  Demonstrations   Protests           Peaceful protest   
8467 2023-04-18  Demonstrations   Protests           Peaceful protest   
8508 2023-04-24  Demonstrations   Protests           Peaceful protest   

                         ACTOR1            ASSOC_ACTOR_1  INTER1  \
0     Protesters (Burkina Faso)  Students (Burkina Faso)       6   
1     Protesters (Burkina Faso)  Students (Burkina Faso)       6   
2     Protesters (Burkina Faso)  Students (Burkina Faso)       6   
3     Protesters (Burkina Faso)  Students (Burkina Faso)       6   
5     Protesters (Burkina Faso)  Students (Burkina Faso)       6   
...                         ...                      ...     ...   
8431     Rioters (Burkina Faso)  Students (Burkina Faso)       5   
8440     Rioters (Burkina Faso)  Students (Burkina Faso)       5   
8453  Protesters (Burkina Faso)                      NaN       6   
8467  Protesters (Burkina Faso)                      NaN       6   
8508  Protesters (Burkina Faso)     Women (Burkina Faso)       6   

                                         ACTOR2 ASSOC_ACTOR_2  INTER2  \
0     Police Forces of Burkina Faso (1987-2014)           NaN       1   
1     Police Forces of Burkina Faso (1987-2014)           NaN       1   
2     Police Forces of Burkina Faso (1987-2014)           NaN       1   
3     Police Forces of Burkina Faso (1987-2014)           NaN       1   
5                                           NaN           NaN       0   
...                                         ...           ...     ...   
8431      Police Forces of Burkina Faso (2022-)           NaN       1   
8440      Police Forces of Burkina Faso (2022-)           NaN       1   
8453                                        NaN           NaN       0   
8467                                        NaN           NaN       0   
8508                                        NaN           NaN       0   

      INTERACTION CIVILIAN_TARGETING        LOCATION  LATITUDE  LONGITUDE  \
0              16                NaN     Ouagadougou    12.370     -1.525   
1              16                NaN     Ouagadougou    12.370     -1.525   
2              16                NaN     Ouagadougou    12.370     -1.525   
3              16                NaN     Ouagadougou    12.370     -1.525   
5              60                NaN     Ouagadougou    12.370     -1.525   
...           ...                ...             ...       ...        ...   
8431           15                NaN     Ouagadougou    12.370     -1.525   
8440           15                NaN     Ouagadougou    12.370     -1.525   
8453           60                NaN  Niankorodougou    10.455     -5.302   
8467           60                NaN          Bittou    11.258     -0.306   
8508           60                NaN            Dori    14.040     -0.030   

                                                  NOTES  FATALITIES  
0     Police fired teargas at protesters demonstrati...           0  
1     Students demontrated for the release of two st...           0  
2     Police fired teargas at student protesters dem...           0  
3     Police fired teargas at students protesting fo...           0  
5     Students protested over circumstances surround...           0  
...                                                 ...         ...  
8431  On 13 April 2

I have to reduce the amount of rows here based on death toll per event. I'm going to look for a date closer to 2014. While there was violence in Burkina Faso before 2014 it was a pretty stable country for 27 years under Blaise Compaore, for better or for worse. I want a date at least a year before the Burkina Faso Uprising that toppled Compaore as the vast majority of political violence in Burkina Faso has occurred since that time, and escalated exponentially only recently. 

In [13]:
# Define the start and end dates of the desired date range
start_date = pd.to_datetime('2013-04-28')
end_date = pd.to_datetime('2023-04-28')

# Filter the dataframe based on the date range
burkina_df2 = burkina_df[(burkina_df['EVENT_DATE'] >= start_date) & (burkina_df['EVENT_DATE'] <= end_date)].copy()

# Reset the index of the new dataframe if desired
burkina_df2.reset_index(drop=True, inplace=True)

In [14]:
burkina_df2.head()

EVENT_DATE       DISORDER_TYPE EVENT_TYPE    SUB_EVENT_TYPE  \
0 2013-05-01      Demonstrations   Protests  Peaceful protest   
1 2013-05-08      Demonstrations   Protests  Peaceful protest   
2 2013-05-13  Political violence      Riots      Mob violence   
3 2013-05-14      Demonstrations   Protests  Peaceful protest   
4 2013-05-15      Demonstrations   Protests  Peaceful protest   

                      ACTOR1  \
0  Protesters (Burkina Faso)   
1  Protesters (Burkina Faso)   
2     Rioters (Burkina Faso)   
3  Protesters (Burkina Faso)   
4  Protesters (Burkina Faso)   

                                       ASSOC_ACTOR_1  INTER1  \
0  CCVC: National Coalition against High Cost of ...       6   
1                                                NaN       6   
2                            Students (Burkina Faso)       5   
3                                                NaN       6   
4                                                NaN       6   

                                      ACTOR2 ASSOC_ACTOR_2  INTER2  \
0                                        NaN           NaN       0   
1                                        NaN           NaN       0   
2  Police Forces of Burkina Faso (1987-2014)           NaN       1   
3                                        NaN           NaN       0   
4                                        NaN           NaN       0   

   INTERACTION CIVILIAN_TARGETING     LOCATION  LATITUDE  LONGITUDE  \
0           60                NaN         Kaya    13.092     -1.084   
1           60                NaN  Ouagadougou    12.370     -1.525   
2           15                NaN  Ouagadougou    12.370     -1.525   
3           60                NaN       Bousse    12.661     -1.892   
4           60                NaN      Oueleni    10.850     -5.350   

                                               NOTES  FATALITIES  
0  CCVC members and others groups staged a march ...           0  
1  Youth organizations staged a march in Ouagadou...           0  
2  Students from high schools in Ouagadougou clas...           0  
3  Protesters removed the keys to the town hall o...           0  
4  Residents from the village of Gorongo stormed ...           0

In [17]:
burkina_df2.sort_values(by='FATALITIES', ascending=False).head(10)

EVENT_DATE       DISORDER_TYPE                  EVENT_TYPE  \
3274 2021-06-04  Political violence  Violence against civilians   
8156 2023-04-20  Political violence  Violence against civilians   
3712 2021-08-18  Political violence                     Battles   
7691 2023-02-17  Political violence                     Battles   
4746 2022-01-19  Political violence                     Battles   
7020 2022-11-13  Political violence                     Battles   
1846 2019-12-24  Political violence                     Battles   
5934 2022-06-11  Political violence  Violence against civilians   
8193 2023-04-27  Political violence                     Battles   
2050 2020-03-10  Political violence                     Battles   

     SUB_EVENT_TYPE                                             ACTOR1  \
3274         Attack       JNIM: Group for Support of Islam and Muslims   
8156         Attack            Military Forces of Burkina Faso (2022-)   
3712    Armed clash                                     Ansaroul Islam   
7691    Armed clash                              Islamic State (Sahel)   
4746    Armed clash        Military Forces of Burkina Faso (2015-2022)   
7020    Armed clash                              Islamic State (Sahel)   
1846    Armed clash  Islamic State (West Africa) - Greater Sahara F...   
5934         Attack                              Islamic State (Sahel)   
8193    Armed clash       JNIM: Group for Support of Islam and Muslims   
2050    Armed clash  Police Forces of Burkina Faso (2015-2022) Gend...   

                                     ASSOC_ACTOR_1  INTER1  \
3274                                           NaN       2   
8156                                           NaN       1   
3712  JNIM: Group for Support of Islam and Muslims       2   
7691                                           NaN       2   
4746             Military Forces of France (2017-)       1   
7020                                           NaN       2   
1846                                           NaN       2   
5934                                           NaN       2   
8193                                           NaN       2   
2050        VDP: Volunteer for Defense of Homeland       1   

                                                 ACTOR2  \
3274                           Civilians (Burkina Faso)   
8156                           Civilians (Burkina Faso)   
3712  Police Forces of Burkina Faso (2015-2022) Gend...   
7691            Military Forces of Burkina Faso (2022-)   
4746  JNIM: Group for Support of Islam and Muslims a...   
7020       JNIM: Group for Support of Islam and Muslims   
1846        Military Forces of Burkina Faso (2015-2022)   
5934                           Civilians (Burkina Faso)   
8193            Military Forces of Burkina Faso (2022-)   
2050  JNIM: Group for Support of Islam and Muslims a...   

                                          ASSOC_ACTOR_2  INTER2  INTERACTION  \
3274                         Labor Group (Burkina Faso)       7           27   
8156                                                NaN       7           17   
3712             VDP: Volunteer for Defense of Homeland       1           12   
7691                                                NaN       1           12   
4746                                                NaN       2           12   
7020                                                NaN       2           22   
1846  Police Forces of Burkina Faso (2015-2022) Gend...       1           12   
5934                                                NaN       7           27   
8193                                                NaN       1           12   
2050                                                NaN       2           12   

      CIVILIAN_TARGETING       LOCATION  LATITUDE  LONGITUDE  \
3274  Civilian targeting         Solhan    13.376      0.457   
8156  Civilian targeting          Karma    13.720     -2.336   
3712                 NaN    Gorol Nyibi    14.

In [30]:
battles = burkina_df2[burkina_df2['EVENT_TYPE']=='Battles']

In [31]:
battles['SUB_EVENT_TYPE'].value_counts()

Armed clash                            1614
Non-state actor overtakes territory      15
Government regains territory              3
Name: SUB_EVENT_TYPE, dtype: int64

In [32]:
battles.head()

EVENT_DATE       DISORDER_TYPE EVENT_TYPE SUB_EVENT_TYPE  \
12 2013-08-31  Political violence    Battles    Armed clash   
15 2013-09-28  Political violence    Battles    Armed clash   
32 2014-01-22  Political violence    Battles    Armed clash   
45 2014-03-28  Political violence    Battles    Armed clash   
46 2014-03-29  Political violence    Battles    Armed clash   

                                         ACTOR1  \
12  Military Forces of Burkina Faso (1987-2014)   
15    Tenkodogo Communal Militia (Burkina Faso)   
32         Fulani Ethnic Militia (Burkina Faso)   
45    Police Forces of Burkina Faso (1987-2014)   
46    Police Forces of Burkina Faso (1987-2014)   

                                  ASSOC_ACTOR_1  INTER1  \
12                                          NaN       1   
15                                          NaN       4   
32                  Pastoralists (Burkina Faso)       4   
45  Military Forces of Burkina Faso (1987-2014)       1   
46  Military Forces of Burkina Faso (1987-2014)       1   

                                               ACTOR2 ASSOC_ACTOR_2  INTER2  \
12  Military Forces of Burkina Faso (1987-2014) Pr...           NaN       1   
15          Tenkodogo Communal Militia (Burkina Faso)           NaN       4   
32                Mossi Ethnic Militia (Burkina Faso)           NaN       4   
45            Unidentified Armed Group (Burkina Faso)           NaN       3   
46            Unidentified Armed Group (Burkina Faso)           NaN       3   

    INTERACTION CIVILIAN_TARGETING      LOCATION  LATITUDE  LONGITUDE  \
12           11                NaN   Ouagadougou    12.370     -1.525   
15           44                NaN     Tenkodogo    11.770     -0.360   
32           44                NaN     Tenkodogo    11.770     -0.360   
45           13                NaN  Fada NGourma    12.062      0.358   
46           13                NaN  Fada NGourma    12.062      0.358   

                                                NOTES  FATALITIES  
12  Explosions and gunfire were heard in Ouagadoug...           1  
15  3injured and 17 arrested during clashes after ...           0  
32  Clashes between Fulani herders and Mossi farme...           1  
45  A joint operations between the security and de...           1  
46  A joint operations between the security and de...           0

Inter1 code 4 is for identity militias. I want to look more into that. 

In [33]:
burkina_df2[burkina_df2['INTER1']==4]

EVENT_DATE       DISORDER_TYPE                  EVENT_TYPE  \
15   2013-09-28  Political violence                     Battles   
32   2014-01-22  Political violence                     Battles   
62   2014-04-26  Political violence  Violence against civilians   
83   2014-07-07  Political violence                     Battles   
189  2015-05-13  Political violence  Violence against civilians   
...         ...                 ...                         ...   
8074 2023-04-09  Political violence  Violence against civilians   
8080 2023-04-10  Political violence                     Battles   
8126 2023-04-16  Political violence                     Battles   
8127 2023-04-16  Political violence                     Battles   
8161 2023-04-22  Political violence                     Battles   

     SUB_EVENT_TYPE                                        ACTOR1  \
15      Armed clash     Tenkodogo Communal Militia (Burkina Faso)   
32      Armed clash          Fulani Ethnic Militia (Burkina Faso)   
62           Attack           Mossi Ethnic Militia (Burkina Faso)   
83      Armed clash  Unidentified Communal Militia (Burkina Faso)   
189          Attack         Zorgo Communal Militia (Burkina Faso)   
...             ...                                           ...   
8074         Attack        VDP: Volunteer for Defense of Homeland   
8080    Armed clash        VDP: Volunteer for Defense of Homeland   
8126    Armed clash        VDP: Volunteer for Defense of Homeland   
8127    Armed clash        VDP: Volunteer for Defense of Homeland   
8161    Armed clash        VDP: Volunteer for Defense of Homeland   

                    ASSOC_ACTOR_1  INTER1  \
15                            NaN       4   
32    Pastoralists (Burkina Faso)       4   
62                            NaN       4   
83         Farmers (Burkina Faso)       4   
189                           NaN       4   
...                           ...     ...   
8074                          NaN       4   
8080                          NaN       4   
8126                          NaN       4   
8127                          NaN       4   
8161                          NaN       4   

                                                 ACTOR2  \
15            Tenkodogo Communal Militia (Burkina Faso)   
32                  Mossi Ethnic Militia (Burkina Faso)   
62                             Civilians (Burkina Faso)   
83         Unidentified Communal Militia (Burkina Faso)   
189                            Civilians (Burkina Faso)   
...                                                 ...   
8074                           Civilians (Burkina Faso)   
8080       JNIM: Group for Support of Islam and Muslims   
8126       JNIM: Group for Support of Islam and Muslims   
8127       JNIM: Group for Support of Islam and Muslims   
8161  JNIM: Group for Support of Islam and Muslims a...   

                                          ASSOC_ACTOR_2  INTER2  INTERACTION  \
15                                                  NaN       4           44   
32                                                  NaN       4           44   
62    Fulani Ethnic Militia (Burkina Faso); Police F...       7           47   
83                          Pastoralists (Burkina Faso)       4           44   
189                         Pastoralists (Burkina Faso)       7           47   
...                                                 ...     ...          ...   
8074                                                NaN       7           47   
8080                                                NaN       2           24   
8126                                                NaN       2           24   
8127                                                NaN       2           24   
8161                                                NaN       2           24   

      CIVILIAN_TARGETING    LOCATION  LATITUDE  LONGITUDE  \
15                   NaN   Tenkodogo    11.770     -0.360   
32                   NaN   Tenkodogo  

In [34]:
burkina_df2.loc[(burkina_df2['INTER1'] == 4) & (~burkina_df2.index.isin(battles.index))]

EVENT_DATE           DISORDER_TYPE                  EVENT_TYPE  \
62   2014-04-26      Political violence  Violence against civilians   
189  2015-05-13      Political violence  Violence against civilians   
191  2015-05-20  Strategic developments      Strategic developments   
277  2016-02-18  Strategic developments      Strategic developments   
662  2018-05-30  Strategic developments      Strategic developments   
...         ...                     ...                         ...   
7789 2023-03-05      Political violence  Violence against civilians   
7796 2023-03-06      Political violence  Violence against civilians   
7835 2023-03-11      Political violence  Violence against civilians   
7886 2023-03-18      Political violence  Violence against civilians   
8074 2023-04-09      Political violence  Violence against civilians   

                      SUB_EVENT_TYPE  \
62                            Attack   
189                           Attack   
191                            Other   
277                            Other   
662                            Other   
...                              ...   
7789                          Attack   
7796  Abduction/forced disappearance   
7835                          Attack   
7886                          Attack   
8074                          Attack   

                                            ACTOR1 ASSOC_ACTOR_1  INTER1  \
62             Mossi Ethnic Militia (Burkina Faso)           NaN       4   
189          Zorgo Communal Militia (Burkina Faso)           NaN       4   
191   Unidentified Communal Militia (Burkina Faso)           NaN       4   
277   Unidentified Communal Militia (Burkina Faso)           NaN       4   
662         Sikire Communal Militia (Burkina Faso)           NaN       4   
...                                            ...           ...     ...   
7789        VDP: Volunteer for Defense of Homeland           NaN       4   
7796        VDP: Volunteer for Defense of Homeland           NaN       4   
7835        VDP: Volunteer for Defense of Homeland           NaN       4   
7886        VDP: Volunteer for Defense of Homeland           NaN       4   
8074        VDP: Volunteer for Defense of Homeland           NaN       4   

                                         ACTOR2  \
62                     Civilians (Burkina Faso)   
189                    Civilians (Burkina Faso)   
191                                         NaN   
277   Police Forces of Burkina Faso (2015-2022)   
662                                         NaN   
...                                         ...   
7789                   Civilians (Burkina Faso)   
7796                   Civilians (Burkina Faso)   
7835                   Civilians (Burkina Faso)   
7886                   Civilians (Burkina Faso)   
8074                   Civilians (Burkina Faso)   

                                          ASSOC_ACTOR_2  INTER2  INTERACTION  \
62    Fulani Ethnic Militia (Burkina Faso); Police F...       7           47   
189                         Pastoralists (Burkina Faso)       7           47   
191                                                 NaN       0           40   
277                                                 NaN       1           14   
662                                                 NaN       0           40   
...                                                 ...     ...          ...   
7789                                                NaN       7           47   
7796  Fulani Ethnic Group (Burkina Faso); Pastoralis...       7           47   
7835                                                NaN       7           47   
7886                                                NaN       7           47   
8074                                                NaN       7           47   

      CIVILIAN_TARGETING   LOCATION  LATITUDE  LONGITUDE  \
62    Civilian targeting       Mane    12.988     -1.344   
189   Civilian targeting      Zorgo    12.249     -0.615   
191 

© 2023 Mickey A. Couvertier

# Change Log
    
| Date (YYYY-MM-DD) | Version | Changed By     | Change Description           |
|-------------------|---------|----------------|------------------------------|
| 2023-05-15        | 1.2     | Mickey A. Couvertier | updated title and body |
| 2023-05-15        | 1.1     | Mickey A. Couvertier | updated map dashboard |
| 2023-05-15        | 1.0     | Mickey A. Couvertier | Added notebook to Github |